File: AngularFrontend/src/app/core/domain-classes/workflow-transition.ts
import { User } from "./user";
import { WorkflowTransitionStatus } from "./workflow-transition-status";
import { WorkflowTransitionRole } from "./workflow-transition-role";
import { WorkflowTransitionUser } from "./workflow-transition-user";

export class WorkflowTransition {
  id?: string;
  workflowId: string;
  name: string;
  fromStepId: string;
  toStepId: string;
  condition: string;
  isFirstTransaction: boolean;
  status?: WorkflowTransitionStatus;
  fromStepName?: string;
  toStepName?: string;
  assignRoles?: string;
  assignUsers?: string;
  completedAt?: Date;
  createdAt?: Date;
  user?: User;
  minutes?: number;
  days?: number;
  hours?: number;
  isUploadDocumentVersion?: boolean;
  isSignatureRequired?: boolean;
  workflowTransitionRoles?: WorkflowTransitionRole[];
  workflowTransitionUsers?: WorkflowTransitionUser[];
  color?: string;
  roleIds?: string[];
  userIds?: string[];
  orderNo?: number;
  comment?: string;
}

File: MySQL-NET-REST-API/DocumentManagement.Data/Entities/WorkflowTransition.cs
using DocumentManagement.Data.Entities;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentManagement.Data;

public class WorkflowTransition
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public Guid WorkflowId { get; set; }
    [ForeignKey("WorkflowId")]
    public Workflow Workflow { get; set; }
    public Guid FromStepId { get; set; }
    [ForeignKey("FromStepId")]
    public WorkflowStep FromWorkflowStep { get; set; }
    public Guid ToStepId { get; set; }
    [ForeignKey("ToStepId")]
    public WorkflowStep ToWorkflowStep { get; set; }
    public string Condition { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public bool IsFirstTransaction { get; set; } = false;
    public int? Days { get; set; }
    public int? Hours { get; set; }
    public int? Minutes { get; set; }
    public bool IsUploadDocumentVersion { get; set; }
    public bool IsSignatureRequired { get; set; }
    public string Color { get; set; }
    public int OrderNo { get; set; }
    public ICollection<WorkflowTransitionRole> WorkflowTransitionRoles { get; set; } = new List<WorkflowTransitionRole>();
    public ICollection<WorkflowTransitionUser> WorkflowTransitionUsers { get; set; } = new List<WorkflowTransitionUser>();
}
